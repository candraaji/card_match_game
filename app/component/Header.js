import React from 'react';
import {View, Text, StyleSheet, Pressable} from 'react-native';

const Header = ({stepToOpen, onPresRestart}) => {
  return (
    <View style={styles.scoreboardContainer}>
      <Pressable testID="restartButton" onPress={onPresRestart}>
        <Text style={styles.textColor}>RESTART</Text>
      </Pressable>
      <Text style={styles.textColor}>STEPS : {stepToOpen}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  scoreboardContainer: {
    minHeight: 50,
    marginTop: '5%',
    justifyContent: 'space-around',
    width: '100%',
    flexDirection: 'row',
  },
  textColor: {
    color: 'white',
  },
});

export default Header;
