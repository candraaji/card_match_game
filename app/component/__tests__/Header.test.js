import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';

describe('Header Component', () => {
  const stepToOpen = 99;
  const onPressRestart = jest.fn();

  test('it should render header compoennt correctly', () => {
    jest.useFakeTimers();
    const wrapper = renderer.create(
      <Header stepToOpen={stepToOpen} onPresRestart={onPressRestart} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
