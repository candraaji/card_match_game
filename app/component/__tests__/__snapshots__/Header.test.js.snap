// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component it should render header compoennt correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-around",
      "marginTop": "5%",
      "minHeight": 50,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="restartButton"
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      RESTART
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "white",
      }
    }
  >
    STEPS : 
    99
  </Text>
</View>
`;
